src/Init/Data/BitVec/Basic.lean

```diff
@@ -116,17 +116,67 @@ end zero_allOnes
 
 section getXsb
 
+/--
+Return the `i`-th least significant bit.
+
+This will be renamed `getLsb` after the existing deprecated alias is removed.
+-/
+@[inline] def getLsb' (x : BitVec w) (i : Nat) (_ : i < w) : Bool := x.toNat.testBit i
```

Comment:
I think it might make sense to bundle the proof, and have this take an `i : Fin w`, akin to how [`List.get`](https://leanprover-community.github.io/mathlib4_docs/Init/Prelude.html#List.get) does it.
IIRC, there are some places we currently have `getLsb i` with `i : Fin w` (relying on the `Fin` to `Nat` coercion). If we bundle the proof in a single `Fin` argument, these call sites will just magically keep working

Reply:
Agreed.